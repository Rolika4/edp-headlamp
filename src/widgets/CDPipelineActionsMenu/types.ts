export interface CDPipelineActionsMenuProps {
    isDetailsPage?: boolean;
}
