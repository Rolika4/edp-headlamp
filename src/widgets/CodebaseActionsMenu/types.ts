export interface CodebaseActionsMenuProps {
    isDetailsPage?: boolean;
}
