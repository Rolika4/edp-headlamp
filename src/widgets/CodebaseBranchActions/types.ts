import { ACTION_MENU_TYPES } from '../../constants/actionMenuTypes';
import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../k8s/EDPCodebaseBranch/types';
import { ValueOf } from '../../types/global';

export interface CodebaseBranchActionsProps {
  data: {
    branch: EDPCodebaseBranchKubeObjectInterface;
    defaultBranch: string;
    codebaseData: EDPCodebaseKubeObjectInterface;
  };
  backRoute?: string;
  variant?: ValueOf<typeof ACTION_MENU_TYPES>;
  anchorEl?: HTMLElement;
  handleCloseResourceActionListMenu?: () => void;
}
