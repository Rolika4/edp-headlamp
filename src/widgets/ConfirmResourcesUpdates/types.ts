import { CRUD_TYPES } from '../../constants/crudTypes';

export interface ResourceToUpdate {
    actionType: CRUD_TYPES;
    kind: string;
    name: string;
}

export interface ConfirmResourcesUpdatesDialogForwardedProps {
    deleteCallback: (...any) => Promise<void>;
    customText?: string;
    resourcesArray: ResourceToUpdate[];
}
