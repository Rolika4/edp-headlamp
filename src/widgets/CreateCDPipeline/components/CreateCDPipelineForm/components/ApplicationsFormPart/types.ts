import { FieldEventTarget, FormNameObject } from '../../../../../../types/forms';

export interface ApplicationsFormPartProps {
    names: {
        [key: string]: FormNameObject;
    };
    handleFormFieldChange(eventTarget: FieldEventTarget): void;
}
