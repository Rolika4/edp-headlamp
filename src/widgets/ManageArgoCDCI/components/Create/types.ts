import { ManageArgoCDIntegrationSecretFormDataContext } from '../../types';

export interface CreateProps {
  formData: ManageArgoCDIntegrationSecretFormDataContext;
}
