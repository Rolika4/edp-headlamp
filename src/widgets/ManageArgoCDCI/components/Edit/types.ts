import { ManageArgoCDIntegrationSecretFormDataContext } from '../../types';

export interface EditProps {
  formData: ManageArgoCDIntegrationSecretFormDataContext;
}
