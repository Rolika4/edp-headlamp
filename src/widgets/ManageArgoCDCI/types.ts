import { SecretKubeObjectInterface } from '../../k8s/Secret/types';
import { FORM_MODES, FormValues } from '../../types/forms';
import { ValueOf } from '../../types/global';
import { ARGOCD_INTEGRATION_SECRET_FORM_NAMES } from './names';

export interface ManageArgoCDIntegrationSecretFormDataContext {
    argoCDSecret: SecretKubeObjectInterface;
    mode: ValueOf<typeof FORM_MODES>;
    ownerReference: string | undefined;
    isReadOnly: boolean;
    handleClosePanel?: () => void;
}

export interface ManageArgoCDIntegrationSecretProps {
    formData: ManageArgoCDIntegrationSecretFormDataContext;
}

export type ManageArgoCDIntegrationSecretFormValues = FormValues<
    typeof ARGOCD_INTEGRATION_SECRET_FORM_NAMES
>;
