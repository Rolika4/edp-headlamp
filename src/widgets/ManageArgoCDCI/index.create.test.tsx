/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { FORM_MODES } from '../../types/forms';
import { ManageArgoCDCI } from './index';

test('renders ManageArgoCDCI Create component', () => {
  render(
    <TestWrapper>
      <ManageArgoCDCI
        formData={{
          argoCDSecret: null,
          argoCDQuickLink: null,
          ownerReference: undefined,
          mode: FORM_MODES.CREATE,
          handleClosePanel: jest.fn(),
        }}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
