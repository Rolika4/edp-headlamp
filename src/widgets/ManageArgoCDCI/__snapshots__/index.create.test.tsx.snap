// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageArgoCDCI Create component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiPaper-root MuiAccordion-root makeStyles-accordion-1 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <span />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                Add ArgoCD Integration
              </h6>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <span
                                  class="makeStyles-labelWrap-3"
                                >
                                  <span
                                    class="MuiTypography-root makeStyles-label-4 MuiTypography-body1"
                                  >
                                    URL
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiInput-input"
                                        name="url"
                                        placeholder="Enter URL"
                                        type="text"
                                        value=""
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <span
                                  class="makeStyles-labelWrap-3"
                                >
                                  <span
                                    class="MuiTypography-root makeStyles-label-4 MuiTypography-body1"
                                  >
                                    Token
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                                        name="token"
                                        placeholder="Enter token"
                                        type="password"
                                        value=""
                                      />
                                      <div
                                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                                      >
                                        <button
                                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                                          tabindex="0"
                                          type="button"
                                        >
                                          <span
                                            class="MuiIconButton-label"
                                          >
                                            <span />
                                          </span>
                                          <span
                                            class="MuiTouchRipple-root"
                                          />
                                        </button>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-space-between"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            cancel
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              <span
                                class="MuiButton-label"
                              >
                                undo changes
                              </span>
                            </button>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              <span
                                class="MuiButton-label"
                              >
                                save
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
